osu.Game.Rulesets.Mania/Mods/ManiaModFadeIn.cs

```diff
@@ -19,5 +29,86 @@ public class ManiaModFadeIn : Mod
         public override double ScoreMultiplier => 1;
         public override bool Ranked => true;
         public override Type[] IncompatibleMods => new[] { typeof(ModFlashlight<ManiaHitObject>) };
+
+        private const float lanecover_size_filled = 0.5f;
+        private const float lanecover_size_gradient = 0.25f;
```

Comment:
don't declare single-use constants. just inlining is fine.

Reply:
> I would explore options as to whether the buffered container can be slotted in solely for the fade-in mod

Adding the buffered container globally was suggested by smoogipoo, so I just went with that. I have no idea how I would even go about inserting it just for FI and HD (though that might just be my inexperience).

> a named field to which the buffered container is assigned

Good idea, will go with that.

> don't declare single-use constants.

I was reluctant about that at first too, but the same thing was done in `CatchModHidden`

Reply:
>Adding the buffered container globally was suggested by smoogipoo

was not aware; in that case I suppose it can stay.

Reply:
I didn't suggest that it should always be here, and agree that it should only be present for FI/HD.

Reply:
As for how to do this, maybe you can do something with the `IApplicableToDrawableRuleset` interface (attach it to your mod). Think you'll probably need to wrap the `hitObjectContainer` in another container and expose it from `HitObjectArea` and then do something like...:
```
var hoc = drawbleRuleset.Playfield.HitObjectArea.HitObjectContainer;
hoc.Parent.Remove(hoc);
hoc.Parent.Add(new BufferedContainer { ... Child = hoc });
```

Reply:
> I didn't suggest that

my bad, misinterpreted then

>you'll probably need to wrap the hitObjectContainer in another container

Doesn't that defeat the purpose of ***not*** wrapping the `hitObjectContainer` globally ?

Reply:
`BufferedContainer` has an overhead, which is the suggestion above is trying to avoid. So without HD/FI the hierarchy will be something like:
```
- Container
  - hitObjectContainer
```
But then with FI/HD it'll become:
```
- Container
  - BufferedContainer
    - hitObjectContainer
```
The overhead of having that `Container` wrapping everything is negligible.